tests/integration/__init__.py

```diff
@@ -108,6 +155,38 @@ def cleanup_runtime_config_instance(to_cleanup):
 
 atexit.register(cleanup_runtime_config_instance, RUNTIME_CONFIGS)
 
+_RUNTESTS_PORTS = {}
+
+
+def get_unused_localhost_port():
+    '''
+    Return a random unused port on localhost
+    '''
+    usock = socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM)
+    usock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    usock.bind(('127.0.0.1', 0))
+    port = usock.getsockname()[1]
+    if port in (54505, 54506, 64505, 64506, 64510, 64511):
+        # These ports are hardcoded in the test configuration
+        usock.close()
+        return get_unused_localhost_port()
```

Comment:
I agree this could be done in another pull request. Like I said earlier, I think the better solution is to simply use reserved ports not in the ephemeral range (which means < 32768). That way, they would never be returned from:

```
usock.bind(('127.0.0.1', 0))
```

I don't think it is a good idea to reserve ports that may be returned from the above command (since any process on the system could do the same and grab one of those ports).

More info on ephemeral ports:
https://en.wikipedia.org/wiki/Ephemeral_port

From the linked article, Linux uses 32768 to 61000 and Windows uses 49152 to 65535, so the safe bet to avoid ephemeral ports on all platforms is to use a number less than 32768.